app/lib/features/pins/widgets/pin_list_item.dart

```diff
@@ -9,11 +10,11 @@ import 'package:go_router/go_router.dart';
 import 'package:acter/common/utils/routes.dart';
 
 class PinListItem extends ConsumerStatefulWidget {
-  final ActerPin pin;
+  final String pinId;
```

Comment:
I understand the rational here, but I don't really agree with the approach taken. It just makes this Widget a lot harder to comprehend and follow. I'd rather we'd have created a wrapper Widget around it do the same:

```dart
class PinListItemById extends ConsumerStatefulWidget {
  final String pinId;
  final bool showSpace;
  const PinListItemById({
    super.key,
    required this.pinId,
    this.showSpace = false,
  });


  Widget build(BuildContext context) {
    return ref.watch(pinProvider(pinId)).when(
      data: (acterPin) => PinListItem(pin: acterPin, showSpace: showSpace),
      loading: () => Skeletonizer(child: Card(child: Column( children: [ ... ]))),
      error: (err, st) => Text('Error loading pin: ${err.toString()}')
    };
}
```